src/shogun/distance/EuclideanDistance.cpp

```diff
@@ -66,11 +77,47 @@ float64_t CEuclideanDistance::compute(int32_t idx_a, int32_t idx_b)
 	return CMath::sqrt(result);
 }
 
+void CEuclideanDistance::precompute_rhs_squared_norms()
+{
+	SGVector<float64_t>rhs_sq=SGVector<float64_t>(rhs->get_num_vectors());
+	
+	for(int32_t idx_i =0; idx_i<rhs->get_num_vectors(); idx_i++)
+	{
+		SGVector<float64_t> tempvec=((CDenseFeatures<float64_t>*) rhs)->get_feature_vector(idx_i);
+		rhs_sq[idx_i]=linalg::dot(tempvec, tempvec);
+		((CDenseFeatures<float64_t>*) rhs)->free_feature_vector(tempvec, idx_i);
+	}
+
+	m_rhs_squared_norms=rhs_sq;
+}
+
+void CEuclideanDistance::precompute_lhs_squared_norms()
```

Comment:
Sorry about joining late :-)

To avoid code duplication, I'd say it is worth to make an auxiliary method that computes the squared norms given the appropriate CDenseFeatures<float64_t> object.